src/Validation/gen/ValidationsGenerator.cs

```diff
@@ -11,28 +14,70 @@ public sealed partial class ValidationsGenerator : IIncrementalGenerator
 {
     public void Initialize(IncrementalGeneratorInitializationContext context)
     {
+        // Get the MSBuild properties for controlling attribute generation
+        var generatorSettings = context.AnalyzerConfigOptionsProvider
+            .Select((configOptions, _) =>
+            {
+                configOptions.GlobalOptions.TryGetValue("build_property.RootNamespace", out var rootNamespace);
+                configOptions.GlobalOptions.TryGetValue("build_property.GenerateBuiltinValidatableTypeAttribute", out var generateAttribute);
+
+                var shouldGenerate = !string.IsNullOrEmpty(rootNamespace) &&
+                                   string.Equals(generateAttribute, "true", StringComparison.OrdinalIgnoreCase);
+
+                return new GeneratorSettings(rootNamespace, shouldGenerate);
+            });
+
+        // Emit the ValidatableTypeAttribute conditionally
+        context.RegisterSourceOutput(generatorSettings, EmitValidatableTypeAttribute);
```

Comment:
This I believe we need to change to `RegisterPostInitializationOuptut` and we can't do it conditionally (we need to always emit the attribute).